### ISSUE TITLE
Change in UI behavior after running nuitka on PyQt (5.11.1) application

### ISSUE THREAD

**Issue by tslator** [kayhayen/Nuitka#132]
Search did not find any existing issues related to PyQt or PyQt 5.11.1

* Please check out if the develop version of Nuitka works better for you.
We are currently using Anaconda, what is the procedure for installing the development version of nuitka?

*  Nuitka version, full Python version and Platform (Windows, OSX, Linux ...)
Nuitka - 0.5.30
Python - 2.7.13
Windows 10

*  How did you install Nuitka and Python (pip, anaconda, deb, rpm, from source,
   what is a virtualenv ...), this is very important usually.
Anaconda

*  If possible please supply a [Short, Self Contained, Correct, Example](http://sscce.org/)
   that demonstrates the issue i.e a small piece of code which reproduces
   the issue and can be run with out any other (or as few as possible)
   external dependencies.
Sorry, not possible

*  If this is a regression (used to work in an earlier version of Nuitka),
   please note what you know about that.
It is regression in that previous versions of the application worked with nuitka but as of about 6 weeks ago, we discovered this issue.

**Comment by brupelo** [comment]
@tslator Hello, at first I had "downvoted" your issue as I thought you were asking something code related and posting the mcve but then I've realized you were actually just asking something more general ;). Next time if it's just a simple question where we won't need any of the template details so the question becomes more clear by doing so just skip it. Anyway, your question being:

> We are currently using Anaconda, what is the procedure for installing the development version of nuitka?

You can refer to the official [download](http://nuitka.net/pages/download.html#id5) docs:

    pip install -U 'http://nuitka.net/gitweb/?p=Nuitka.git;a=snapshot;h=refs/heads/develop;sf=tgz'

that should do, if that doesn't work with you, just:

    - `git clone https://github.com/kayhayen/Nuitka && cd Nuitka`
    - pip install -e .
    - `git checkout <the-branch-you-want-to-use>`

But if you just want to use nuitka casually, I'd strongly recommend you to follow the official download docs as those ones are already providing nicely some packages ready to be consumed by users. That will make easier bug reporting, helping and so on... ;)

**Comment by kayhayen** [comment]
@brupelo Please never downvote anything, I do not welcome that, even if it were appropriate, which it is not. I do not want this to be an unwelcoming place. And do not discourage from filling out the issue template, the reporters can and do never know, and often it helps me. 

@tslator Although develop version will be substantially different from stable, there is no need to try it out though. In this instance, you reported using an older version. Did you try the latest stable version, updating via conda there? It is relatively old. Or did you mean to say that was the version still working, and the recent one is not?

Would it be possible for you to "git bisect", or is that going to be too difficult for you? In that you would checkout a good state of Nuitka from git, the version that works, tag it as good, then the recent version, try it and tag it as bad, then git picks version between, you test.

Depending on the range, it takes a few tries, but not a lot, typically just a couple. If you cannot give us a minimal reproducer, fine if you do the git bisect. Otherwise there is no way forward @tslator as how will I know anything? Given the commit that changed it for you, I can at least critically examine how it would cause something and provide you with instructions to see the delta effect in generated code. 

@brupelo I wonder where to put debugging techniques for users to use. I suppose User Manual should be a good place for that. So instead of giving instructions, I want to have pre-canned text there, and point to it, with links. Debugging standalone ImportError also has that. Of course only stuff, users can actually do.

**Comment by brupelo** [comment]
@brupelo Please never downvote anything, I do not welcome that, even if it were appropriate, which it is not. I do not want this to be an unwelcoming place. And do not discourage from filling out the issue template, the reporters can and do never know, and often it helps me.

Makes total sense, my bad. Thanks to clarify! ;)

@brupelo I wonder where to put debugging techniques for users to use. I suppose User Manual should be a good place for that. So instead of giving instructions, I want to have pre-canned text there, and point to it, with links. Debugging standalone ImportError also has that. Of course only stuff, users can actually do.

Your call, few remarks here, as I told in other threads, at the end of the day, try to be very simplistic when it comes to provide casual-user docs, period:

 - Make the docs as simple as possible and try get rid of irrelevant details. Give them little text for to read (I usually favour cheatsheets when it comes to pypi packages usage)
 - Put more effort writing about specific casual user use-cases, that's what they're interested about (cool examples so you'll attract their attention), agressive marketing claims about those use-cases are not a bad thing either, at least at the beginning when we want to get traction with the project ;)
 - Summing up: Simplify their lives and try to engage them ;) 

Plus, chances casual users will become dev-contributors is usually very low (maybe i'm wrong about this one)... but if they get interested they will definitely help us by providing more bug reports.

Of course, what's exposed above doesn't apply to developers, in that case I guess proper documentation is very relevant (I like the one who's living in the official site)... and don't get me wrong, when I say proper documentation I don't mean breaking the DRY principle, to see what i mean, [here](https://softwareengineering.stackexchange.com/a/300061/242854)

**Comment by kayhayen** [comment]
@tslator Any news on this?

**Comment by tslator** [comment]
Hi @kayhayen 

Sorry for the late response.  To clarify, we have been using the same nuitka version.  What we recently observed was a change in our application behavior.  Admittedly, it is possible this behavior has always been there but wasn't detected or that recent application changes are contributing to the issue.  We know that this behavior does not occur when running from source and only occurs in the nuitka output.  I think the best course of action is to test with the latest stable release of nuitka.

**Comment by tslator** [comment]
That latest release in CondaForge for win64 is 0.5.30 which is what we are using.  The latest noarch release is 0.5.33.  Can I use the noarch release on windows?

**Comment by kayhayen** [comment]
Yes you can, Tim, in fact Nuitka was always "noarch" and they were packaging it wrongly. Definitely should use that release, there have been many fixes. I also think for the PyQt plugins.

Let me know if the your issue persists be re-opening this. However, without a reproducer, there is only so much that can be done.
